include/circt/Dialect/SMT/SMTBitVectorOps.td

```diff
@@ -0,0 +1,54 @@
+//===- SMTBitVectorOps.td - SMT bit-vector dialect ops -----*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_SMT_SMTBITVECTOROPS_TD
+#define CIRCT_DIALECT_SMT_SMTBITVECTOROPS_TD
+
+include "circt/Dialect/SMT/SMTDialect.td"
+include "circt/Dialect/SMT/SMTAttributes.td"
+include "circt/Dialect/SMT/SMTTypes.td"
+include "mlir/IR/EnumAttr.td"
+include "mlir/IR/OpAsmInterface.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+class SMTBVOp<string mnemonic, list<Trait> traits = []> :
+  Op<SMTDialect, "bv." # mnemonic, traits>;
+
+def BVConstantOp : SMTBVOp<"constant", [
+  Pure,
+  ConstantLike,
+  FirstAttrDerivedResultType,
+  DeclareOpInterfaceMethods<InferTypeOpInterface, ["inferReturnTypes"]>,
+  DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>
+]> {
+  let summary = "produce a constant bit-vector";
+  let description = [{
+    This operation produces an SSA value equal to the bit-vector constant
+    specified by the 'value' attribute.
+    Refer to the `BitVectorAttr` documentation for more information about
+    the semantics of bit-vector constants, their format, and associated sort.
+    The result type always matches the attribute's type.
+
+    Examples:
+    ```mlir
+    %bv_x5c = smt.bv.constant <"#x5c">
+    %bv_b0101 = smt.bv.constant <"#b0101">
```

Comment:
This exact format is only possible with a custom parser and printer for the constant op. I'd rather just use the default `BitVectorAttr` syntax directly, i.e., `#smt.bv<92> : !smt.bv<16>` to avoid implementing an ugly parser.
Since we need the type when using decimal numbers (because it cannot be inferred from just the constant) and want to reuse the type parsed by the MLIR infrastructure for `TypedAttr`s, we need to use `quantified($value)`, i.e., it's not possible to shorten it to `<92> : !smt.bv<16>`. The angle brackets around the 92 (or some other punctuation) are always necessary because the parser cannot deal with spaces in the extended syntax at this location.

Reply:
That makes sense to me, but I'm wondering if I would be prettier with the `qualified` format, to not have weird angle brackets there.

Reply:
Yes, I went with the `qualified,` also because only then is the type parsed and printed by the MLIR infra.